internal/app/networkd/pkg/nic/nic.go

```diff
@@ -0,0 +1,63 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package nic
+
+import (
+	"errors"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/talos-systems/talos/internal/app/networkd/pkg/address"
+)
+
+// TODO: Probably should put together some map here
+// to map number with string()
+const (
+	Bond = iota
+	Single
+
+	// https://tools.ietf.org/html/rfc791
+	MinimumMTU = 68
+	MaximumMTU = 65536
+)
+
+// NetworkInterface provides an abstract configuration representation for a
+// network interface
```

Comment:
As a general rule of thumb, Public funcs/methods require a comment, and we should comment with proper grammar (ending in a period in this case) since this ends up the public GoDoc site. More of a note for the future.